staging/src/k8s.io/apiserver/pkg/server/options/serving_with_loopback.go

```diff
@@ -79,3 +127,34 @@ func (s *SecureServingOptionsWithLoopback) ApplyTo(secureServingInfo **server.Se
 
 	return nil
 }
+
+// NewLoopbackClientCert creates a self-signed certificate and key for the loopback client.
+func (s *SecureServingOptionsWithLoopback) NewLoopbackClientCert() ([]byte, []byte, error) {
+	certPem, keyPem, err := certutil.GenerateSelfSignedCertKey(server.LoopbackClientServerNameOverride, nil, nil)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate self-signed certificate for loopback connection: %v", err)
+	}
+	return certPem, keyPem, nil
+}
+
+// NewLoopbackClientConfig creates a loopback client *rest.Config for the given token and certificate.
```

Comment:
@sttts note, this is not 100% identical to `(*secureServingInfo).NewLoopbackClientConfig()`. That function requires an active `Listener`. Should I try to refactor so we don't have duplicate similar methods?

Reply:
Good point... If it's 0, we can't create the client config until after the server has started, which won't work for kcp. I can have it return an error if the port is 0, and make the godoc explicit that it doesn't support 0, at least for now?

Reply:
yes, error + godoc is fine.

Reply:
New commit added - PTAL